### ISSUE TITLE
Search is very slow, especially for tags that are rarely used

### ISSUE THREAD

**Issue by StashCat** [tjg1/nori#91]
What I believe this does is loop through ALL submissions and check what tags they have, instead of the site's own search function which is much faster.

**Comment by tjg1** [comment]
Yes, norilib does iterate through all the images to implement filtering. It could definitely use more optimisation - I think at the moment we have separate loops for tags and safety ratings.

It would be great to implement a benchmark suite to see if any changes make a significant difference.

Either that or output timing information to Logcat in DEBUG builds. The format could be something like "Network took 121ms. Parsing took 50ms. Filtering took 75ms."
